### PR TITLE
OCPCLOUD-2191: Update to Kube 1.28

### PR DESCRIPTION
**Pull request by nrb**
Also update the build image to use golang 1.20

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,11 +11,11 @@ require (
...
-	k8s.io/apimachinery v0.27.4
-	k8s.io/client-go v0.27.2
+	golang.org/x/oauth2 v0.8.0
+	google.golang.org/api v0.114.0
+	k8s.io/api v0.28.2
+	k8s.io/apimachinery v0.28.2
+	k8s.io/client-go v0.28.2
 	k8s.io/klog/v2 v2.100.1
 	k8s.io/utils v0.0.0-20230726121419-3b25d923346b
 	sigs.k8s.io/controller-runtime v0.15.0
```
Comment (damdo) [inline go.mod:21]: You may also want to bump the controller-runtime to the latest

**Comment by damdo** [comment]
The import linting is not happy 🤔

**Comment by nrb** [comment]
Yeah, I let GoLand do it which doesn't match OpenShift's ordering.

I've fixed that, still debugging the termination handler failing

**Review by damdo — COMMENTED** [review]
/approve

**Pull request merged**